java-squid/src/main/java/org/sonar/java/se/SymbolicValue.java

```diff
@@ -163,13 +165,20 @@ public void computedFrom(List<SymbolicValue> symbolicValues) {
     }
 
     @Override
-    public List<ProgramState> setConstraint(ProgramState programState, BooleanConstraint booleanConstraint) {
+    public List<ProgramState> setConstraint(ProgramState initialProgramState, BooleanConstraint booleanConstraint) {
+      ProgramState programState = initialProgramState;
       if (leftOp.equals(rightOp)) {
         if (shouldNotInverse().equals(booleanConstraint)) {
           return ImmutableList.of(programState);
         }
         return ImmutableList.of();
       }
+      if (leftOp.id > 2 && rightOp.id > 2) {
```

Comment:
This is a bad smell... it should be first factorized in a dedicated method to name what this condition does and then there is probably more thinking to be done to solve the problem of this exclusion in a more elegant manner.

IMO we should not end up differentiating equality to null/true/false and to other SV. Relation between SV should be a generalization of what was done on null constraints.